nucypher/blockchain/eth/agents.py

```diff
@@ -814,29 +819,21 @@ def create_policy(self,
         return receipt
 
     @contract_api(CONTRACT_CALL)
-    def fetch_policy(self, policy_id: str, with_owner=False) -> Union[tuple, list]:
+    def fetch_policy(self, policy_id: bytes) -> Policy:
```

Comment:
was there a reason we needed to change the type from `str` to `bytes`  for the `policy_id`?

Reply:
Presumably, the contract API always took bytes i.e. from before and would not be due to new code ...

I only ask because I think we've tried to limit the surface area passing bytes around.

Reply:
My impression is that the hint for `str` was just incorrect.  The `PolicyManager` itself contract takes `bytes16`.  Using a string encoding and format actually seems more complex to me.

Reply:
👍